flink-core/src/main/java/org/apache/flink/core/failure/FailureEnricher.java

```diff
@@ -124,5 +140,16 @@ enum FailureType {
          * @return the Executor pool
          */
         Executor getIOExecutor();
+
+        /**
+         * Get the meta information of current job.
+         *
+         * @return the job meta information.
+         */
+        @PublicEvolving
+        @Nullable
+        default JobInfo getJobInfo() {
```

Comment:
Same here for default and nullable.

Reply:
I've changed the default methods including `getJobInfo()` and `getTaskInfo()` to be abstract.